### PR TITLE
feat: realistic positions in localosmosis with state

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

This PR adds realistic positions to localosmosis by querying Ethereum's Uniswap WETH / USDC pool and converting it into Osmosis genesis.

It does so by introducing a Go script. This script has 3 commands set via flags:
1. `getData` retrieves the data from the Uniswap subgraph and writes it to the disk
2. `convertPositions` converts the data from the Uniswap subgraph into Osmosis genesis by reading the file from the previous step and spinning up a mock Osmosis app to create positions on and, then, exporting genesis
3. `mergeSubgraphAndLocalOsmosisGenesis` merges the genesis created from the subgraph data with the localosmosis genesis. It is meant to only be called inside a localosmosis container

There are makefile commands defined that should help to interact with the script on the user's host machine.

## LocalOsmosis

 These commands are also executed in `setup.sh` when starting up a localosmosis instance with `make localnet-start-with-state`

## Brief Changelog

- implement the script
- implement makefile commands
- integrate it into localosmosis

## Testing and Verifying

- tested each makefile step individually
- tested swapping on localosmosis and querying

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

**Review by mattverse — COMMENTED** [review]
Where can I find this pool info in info.uniswap.org or uniswap interface?

Can't seem to find WETH in https://info.uniswap.org/#/tokens top token lists as well. Would it make sense to maybe use ETH / USDC pool instead?

File: tests/cl-genesis-positions/convert.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+		ExponentAtPriceOne: sdk.OneInt().Neg(),
+		SwapFee:            sdk.MustNewDecFromStr("0.0005"),
+	}
+
+	poolId, err := osmosis.App.PoolManagerKeeper.CreatePool(osmosis.Ctx, msgCreatePool)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(poolId)
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply tests/cl-genesis-positions/convert.go:117]: https://www.geckoterminal.com/eth/pools/0x88e6a0c2ddd26feeb64f039a2c41296fcb3f5640

**Comment by p0mvn** [comment]
> Where can I find this pool info in info.uniswap.org or uniswap interface?
> 
> Can't seem to find WETH in https://info.uniswap.org/#/tokens top token lists as well. Would it make sense to maybe use ETH / USDC pool instead?

https://www.geckoterminal.com/eth/pools/0x88e6a0c2ddd26feeb64f039a2c41296fcb3f5640

**Comment by mattverse** [comment]
Ah its because uniswap interfaces show WETH as ETH, undertsood

File: tests/cl-genesis-positions/get_data.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	"sort"
+)
+
+func GetUniV3SubgraphData(pathToSaveAt string) {
+	// Set the subgraph URL and query.
+	const (
+		subgraphURL = "https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3"
+		query       = `{
+			positions(
+				first: 1000
```
Comment (mattverse) [inline tests/cl-genesis-positions/get_data.go:18]: asking to learn: it seems that this makes it return batches of 1000, what is it that we are batching? The Positions or the ticks?

**Review by mattverse — APPROVED** [review]
utACK

File: tests/cl-genesis-positions/get_data.go (hunk 2 above)
Comment (p0mvn) [reply tests/cl-genesis-positions/get_data.go:18]: The positions. 1000 is the max number it is possible to get per query

**Pull request merged**